pkg/docker/config/config_test.go

```diff
@@ -886,3 +886,92 @@ func TestValidateKey(t *testing.T) {
 		assert.Equal(t, tc.isNamespaced, isNamespaced)
 	}
 }
+
+func TestSetGetCredentials(t *testing.T) {
+	const (
+		username = "username"
+		password = "password"
+	)
+
+	getAuth := func(sys *types.SystemContext, input string) types.DockerAuthConfig {
+		ref, err := reference.ParseNamed(input)
+		require.NoError(t, err)
+		auth, err := GetCredentialsForRef(sys, ref)
+		require.NoError(t, err)
+		return auth
+	}
+
+	// Unset the home dir to make the test not fallback to other auth files
+	// there.
+	tmpDir, err := ioutil.TempDir("", "auth-test-")
+	require.NoError(t, err)
+	os.Setenv("HOME", tmpDir)
+	defer os.RemoveAll(tmpDir)
+
+	for _, tc := range []struct {
+		name       string
+		set        string
+		get        string
+		setByAPI   bool
```

Comment:
Absolutely non-blocking: I’m not sure we need this; just testing the lookup finds the right entries in a fixture might test the lookup path enough. Compare `TestGetAuth`.

Alternatively test both a fixture, and the set/lookup API round-trip.

But, sure, these tests are already infinitely better than the previous no tests.

Reply:
Removed that one and always stick to create a file by coinstructing a plain string.